### PR TITLE
fix: reduce cloning for dedupe

### PR DESCRIPTION
**Pull request by laststylebender14**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #... _(Replace "..." with the issue number)_

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/core/jit/graphql_executor_arc.rs [hunk 1]
```diff
@@ -127,16 +129,45 @@ impl JITArcExecutor {
...
+        async_graphql::Response::default().into()
+    }
+}
+
+impl From<async_graphql::Response> for ByteResponse {
+    fn from(mut response: async_graphql::Response) -> Self {
+        ByteResponse {
+            cache_control: std::mem::take(&mut response.cache_control),
+            is_ok: response.errors.is_empty(),
+            data: serde_json::to_vec(&response).unwrap(),
```
Comment (laststylebender14) [inline src/core/jit/graphql_executor_arc.rs:156]: fix the unwrap

File: src/core/http/request_handler.rs [hunk 2]
```diff
@@ -73,6 +73,19 @@ fn update_cache_control_header(
     response
 }
 
+fn update_cache_control_header_for_arc(
```
Comment (karatakis) [inline src/core/http/request_handler.rs:76]: Can you add this in impl, so you can chain the functions latter

File: src/core/http/request_handler.rs [hunk 3]
```diff
@@ -134,21 +147,20 @@ async fn execute_query<T: DeserializeOwned + GraphQLRequestLike>(
...
-        request
-            .execute(&JITExecutor::new(
-                app_ctx.clone(),
-                req_ctx.clone(),
-                operation_id,
-            ))
-            .await
+        let exec = JITExecutor::new(app_ctx.clone(), req_ctx.clone(), operation_id);
+        let mut response = request.execute_with_jit(exec).await;
+        response = update_cache_control_header_for_arc(response, app_ctx, req_ctx.clone());
```
Comment (karatakis) [inline src/core/http/request_handler.rs:152]: use functionchaining to increase readability

File: src/core/http/request_handler.rs [hunk 4]
```diff
@@ -134,21 +147,20 @@ async fn execute_query<T: DeserializeOwned + GraphQLRequestLike>(
...
+
+        response.into_response()?
     } else {
-        request.data(req_ctx.clone()).execute(&app_ctx.schema).await
+        let mut response = request.data(req_ctx.clone()).execute(&app_ctx.schema).await;
+        response = update_cache_control_header(response, app_ctx, req_ctx.clone());
```
Comment (karatakis) [inline src/core/http/request_handler.rs:157]: use function chaining to increase readability

**Review by karatakis — APPROVED** [review]
OK

**Pull request merged**